src/python/pants/testutil/rule_runner.py

```diff
@@ -342,8 +344,19 @@ def add_to_build_file(
         mode = "w" if overwrite else "a"
         return self.create_file(str(build_path), target, mode=mode)
 
+    def write_files(self, files: Mapping[str, str]) -> None:
```

Comment:
It would be nice to let `str | PurePath` punch through to the key type.

Reply:
Good idea. To save trees, I'll add in follow-up I'm working on.